vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -1,126 +1,282 @@
 """The data manager handles access to all DataFrames used in a Vizro app."""
 
-from typing import Callable, Dict, Union
+from __future__ import annotations
```

Comment:
I am going to refactor this stuff in the next PR where I remove `get_component_data`, and that should make it clearer where these calls are coming from. Let me leave this conversation unresolved to remind me about it 🙂 

With that said, I don't think multiple calls to "Loading data" affects response time significantly, because the whole point of the cache is that it's quick to retrieve from (I guess `FileSystemCache` is slower than the others though), e.g. if caching is `SimpleCache` then it should really add essentially zero time because the dataframe is just in the Python process memory. Do you have any profiling to suggest that this really does slow things down?

Reply:
Only the fact we retrieve the same data source twice in a single API call, tells that something isn't totally fine-tuned. It affects response time significantly if no cache is set (which is one of the production-ready solutions we provide and mention in the docs) and where the static data is used and referenced by name. eg:
```
def get_api_data():
    response = requests.get('https://api.example.com/data')
    api_data = response... # to pd.DataFrame conversion 
    return api_data

data_manager["api_df"] = get_api_data
...
vm.Graph(figure=px.scatter(data_frame="api_df", "sepal_width", "sepal_length"), ...),
...
```

It should be straight-forward to get rid of this double graph/table/ag_grid creation, so let's keep this one opened as you said. 😄

Reply:
Ah yes, of course this is obviously important if we don't have cache turned on, which as you say is the default behaviour for dynamic data.

Very good observation! I will need to make some follow up tickets based on this PR anyway so will make sure I include this.